### PR TITLE
[accelerate] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to accelerate.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

1. I am looking for any `!missing-null-allowed!` in this frameworks ignore files and applying them
2. I am adding `nullable enable` to all manual files that are not "API_SOURCES" in src/frameworks.sources and making the required nullability changes
3. Changing all `throw new ArgumentNullException ("object"));` to `ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (object));` for size saving optimization as well to mark that this framework contains nullability changes
4. Changing any `== null` or `!= null` to `is null` and `is not null`

### CODE REVIEW

File: src/Accelerate/vImageTypes.cs [hunk 1]
```diff
@@ -272,11 +272,11 @@ public static vImageError ConvolveWithBiasARGBFFFF (ref vImageBuffer src, ref vI
 									vImageFlags flags)
 		{
 			if (kernels == null)
```
Comment (spouliot) [inline src/Accelerate/vImageTypes.cs:274]: you can also update the `== null` for `is null`
and `!= null` for `is not null`

That was not available when I wrote https://github.com/xamarin/xamarin-macios/wiki/Nullability but it should be added :)

Comment (spouliot) [reply src/Accelerate/vImageTypes.cs:274]: > Changing any == null or != null to is null and is not null

well well well :D

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
Please make the requested change from @spouliot

File: src/Accelerate/vImageTypes.cs (hunk 1 above)
Comment (tj-devel709) [reply src/Accelerate/vImageTypes.cs:274]: I wrote down that this PR does the 'is null' and 'is not null' but I guess I lied haha. Thank you for catching my bluff

**Comment by tj-devel709** [comment]
@spouliot and @mandel-macaque I did the 'is null' and 'is not null' changes as well as updated that wiki page to include it!

**Review by dalexsoto — APPROVED** [review]

File: src/Accelerate/vImageTypes.cs [hunk 2]
```diff
@@ -271,12 +271,12 @@ public static vImageError ConvolveWithBiasARGBFFFF (ref vImageBuffer src, ref vI
 									Pixel8888 backgroundColor,
```
Comment (rolfbjarne) [inline src/Accelerate/vImageTypes.cs:271]: This file isn't in `API_SOURCES`, so according to your PR description it should get `#nullable enable` added

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/Accelerate/vImageTypes.cs (hunk 2 above)
Comment (tj-devel709) [reply src/Accelerate/vImageTypes.cs:271]: Ah good catch, thank you!

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
Don't land this, I want confirmation that we are ok by throwing new diff exceptions when null in certain cases.

File: src/Accelerate/vImageTypes.cs [hunk 3]
```diff
@@ -437,12 +439,19 @@ public static vImageError TentConvolveARGB8888 (ref vImageBuffer src, ref vImage
...
+			if (pre_bias is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (pre_bias));
+			else if (pre_bias.Length != 4)
 				throw new ArgumentException ("Must have four elements", "pre_bias");
-			if (post_bias != null && post_bias.Length != 4)
+
+			if (post_bias is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (post_bias));
+			else if (post_bias.Length != 4)
 				throw new ArgumentException ("Must have four elements", "post_bias");
```
[the remark this answers is not in the document]
Comment (chamons) [reply src/Accelerate/vImageTypes.cs:453]: We should not change behavior in a PR like this. If we decide we want to change behavior, it should be either in another PR or at least in a separate commit in this asking for special review. 

These PRs are mechanical enough, don't let non-mechanical changes hide in the diff if possible.

[the remark this answers is not in the document]
Comment (rolfbjarne) [reply src/Accelerate/vImageTypes.cs:453]: In any case this change is wrong, the P/Invoke accepts NULL for the parameter in question (which means that the managed parameter should become nullable instead).

[the remark this answers is not in the document]
Comment (tj-devel709) [reply src/Accelerate/vImageTypes.cs]: @mandel-macaque @chamons @rolfbjarne Reverting these changes here! Going to hop on a little call later with Manuel to make sure I understand what I did wrong up there and hopefully don't do in future PRs. Thanks for the review!

**Review by rolfbjarne — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**